frontend/src/lib/errorUtils.ts

```
@@ -0,0 +1,313 @@
+  switch (errorType) {
+    case "network":
+      return "Connection failed. Please check your internet connection and try again.";
+
+    case "permission":
+      if (status === 401) {
+        return "Your session has expired. Please log in again.";
+      }
+      return "You do not have permission to perform this action.";
+
+    case "validation":
+      const message = formatError(error);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)